From 3ec19a923a478fc6788d7b094d34cf46189d9d6c Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Thu, 8 Jan 2015 14:53:29 +0800
Subject: [PATCH 0368/1594] MLK-10078-04 arm: mcc: enable mcc tests in dts

commit 6e2cf20f796476a83fdc3f12c890c80191d65e43 from
git://git.freescale.com/imx/linux-2.6-imx.git

based on imx_3.14 kernel, add mcc pingpong and tty tests
on imx6sx sdb and ai boards.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto-m4.dts |   15 +++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb-m4.dts       |   15 +++++++++++++++
 arch/arm/boot/dts/imx6sx.dtsi             |   10 ++++++++++
 3 files changed, 40 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
index 436388b..d6743cb 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
@@ -8,6 +8,13 @@
 
 #include "imx6sx-sabreauto.dts"
 
+/{
+	memory {
+		linux,usable-memory = <0x80000000 0x3ff00000>;
+		reg = <0x80000000 0x40000000>;
+	};
+};
+
 /*
  * The flollowing modules are conflicting with M4, disable them when m4
  * is running.
@@ -24,6 +31,14 @@
         status = "disabled";
 };
 
+&mcctest{
+	status = "okay";
+};
+
+&mcctty{
+	status = "okay";
+};
+
 &uart2 {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6sx-sdb-m4.dts b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
index e5b312a..5309be9 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
@@ -8,6 +8,13 @@
 
 #include "imx6sx-sdb.dts"
 
+/{
+	memory {
+		linux,usable-memory = <0x80000000 0x3ff00000>;
+		reg = <0x80000000 0x40000000>;
+	};
+};
+
 /*
  * The flollowing modules are conflicting with M4, disable them when m4
  * is running.
@@ -24,6 +31,14 @@
         status = "disabled";
 };
 
+&mcctest{
+	status = "okay";
+};
+
+&mcctty{
+	status = "okay";
+};
+
 &uart2 {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 4f54eb7..78715f5 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1354,6 +1354,16 @@
 				status = "okay";
 			};
 
+			mcctest: mcctest{
+				compatible = "fsl,imx6sx-mcc-test";
+				status = "disabled";
+			};
+
+			mcctty: mcctty{
+				compatible = "fsl,imx6sx-mcc-tty";
+				status = "disabled";
+			};
+
 			uart6: serial@022a0000 {
 				compatible = "fsl,imx6sx-uart", "fsl,imx21-uart";
 				reg = <0x022a0000 0x4000>;
-- 
1.7.5.4

